src/content/en/2021/caching.md

```diff
@@ -17,4 +20,406 @@ featured_stat_3: TODO
+    </tr>
+    <tr>
+     <td><code class="no-wrap">private</code></td>
+     <td>The response is intended for a specific user and should not be stored by shared caches such as CDNs.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">immutable</code></td>
+     <td>Indicates that the cached entry will never change during its TTL and that revalidation is not necessary.</td>
+    </tr>
+  </table>
+  <figcaption>TODO: Caption the table above</figcaption>
+</figure>
+
+The `max-age` directive is the most commonly found since it directly defines the TTL the same way that the `Expires` header does.
+
+{{ figure_markup(
+  image="cache-control-directives.png",
+  caption="Usage of `Cache-Control` directives.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1359015817&format=interactive",
+  sheets_gid="1944529311",
+  sql_file="cache_control_directives.sql"
+) }}
+
+Cache Control directives, 61% of mobile requests include a Cache-Control response header with a `max-age` directive.
+
+The most misused Cache Control directive is `set-cookie`, used for 0.07% of total directives for Desktop and 0.08% for Mobile.
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+### `304` Not Modified status
+
+{{ figure_markup(
+  image="http-304-by-caching-strategy.png",
+  caption="HTTP 304 response rate by caching strategy.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1965033294&format=interactive",
+  sheets_gid="1136058277",
+  sql_file="valid_if_none_match_returns_304.sql"
+) }}
+
+### Validity of date strings
+
+The three main HTTP headers used to represent timestamps, `Date`,`Last-Modified` and `Expires` all use a date formatted string. The `Date` HTTP response header is almost always generated automatically by the web server, meaning that invalid values are extremely rare. Still, in the rare event that the date is set incorrectly it can affect cacheability on the response on which it is served. 
+
+[TODO Add figure showing on Mobile Invalid date strings between pct_using_invalid_date, pct_using_invalid_last_modified and  pct_using_invalid_expires, data from 2020 https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/edit#gid=1338572274 and 2021 https://docs.google.com/spreadsheets/d/1-v3yR0LZIC3t4zWtqTgR3jJsKjjRMP-HATU2caP8e2c/edit#gid=1680471251 Potentiall we mighe want to show 2020 vs 2021]
```

Comment:
cc @rviscomi

Reply:
Focusing only on 2021 data for the figures by default. If you wanted to draw a comparison to 2020 you could include a link in the paragraph to the 2020 figure.

```suggestion
{{ figure_markup(
  image="invalid-date-formats.png",
  caption="Percent of responses with invalid date formats.",
  description="TODO",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=383751268&format=interactive",
  sheets_gid="1680471251",
  sql_file="invalid_last_modified_and_expires_and_date.sql"
) }}
```